server/topic.go

```diff
@@ -2463,8 +2454,9 @@ func (t *Topic) replyGetDel(sess *Session, asUid types.Uid, req *MsgGetOpts, msg
 		return errors.New("invalid MsgGetOpts query")
 	}
 
-	// Check if the user has permission to read the topic data and the request is valid
-	if userData := t.perUser[asUid]; (userData.modeGiven & userData.modeWant).IsReader() || asChan {
+	// Check if the user has permission to read the topic data and the request is valid.
+	// Don't permit channel readers access deleted messages because they cannot delete messages.
```

Comment:
Channel readers cannot delete messages but should we show them gaps/placeholders for deleted messages?

Reply:
This is a good point. I don't think channel readers should see the gaps, but they should be able to remove messages deleted by the publisher. I'll enable {get what=del} for channel readers. Special handling of gaps must be implemented on the client.